src/main/java/org/testng/TestRunner.java

```diff
@@ -254,6 +254,9 @@ private void init() {
 
     initListeners();
     addConfigurationListener(m_confListener);
+    for (IConfigurationListener cl : m_configuration.getConfigurationListeners()) {
```

Comment:
the other listeners work just fine, this PR is just handling the missing configuration listener.

this need to be trace back to where the configuration listener gets notified, `org.testng.internal.Invoker.runConfigurationListeners(ITestResult, boolean)`:
```java
  private void runConfigurationListeners(ITestResult tr, boolean before) {
    if (before) {
// Nick: m_notifier is the instance of TestRunner
      for(IConfigurationListener icl: m_notifier.getConfigurationListeners()) {
        if (icl instanceof IConfigurationListener2) {
          ((IConfigurationListener2) icl).beforeConfiguration(tr);
        }
      }
    } else {
      for(IConfigurationListener icl: m_notifier.getConfigurationListeners()) {
        switch(tr.getStatus()) {
          case ITestResult.SKIP:
            icl.onConfigurationSkip(tr);
            break;
          case ITestResult.FAILURE:
            icl.onConfigurationFailure(tr);
            break;
          case ITestResult.SUCCESS:
            icl.onConfigurationSuccess(tr);
            break;
        }
      }
    }
  }
```

TestNG Eclipse plugin uses testng-remote, where create the TestRunner instance (see corresponding fix https://github.com/testng-team/testng-remote/pull/39), but we can see the user custom configuration listener was not added to TestRunner's m_configurationListeners.
I think it's TestRunner's responsibility to add the configuration listener which passed from TestNG, rather than expose the logic and let the downstream consumer to handle this init task.